pkgs/development/compilers/gcc/common/extra-target-flags.nix

```diff
@@ -9,8 +9,11 @@ in
   # It would be good to consolidate the generation of makeFlags
   # ({C,CXX,LD}FLAGS_FOR_{BUILD,TARGET}, etc...) at some point.
   EXTRA_FLAGS_FOR_TARGET = let
+      commonFlags = "-O2";
+      nanoFlags = "-g -Os -fdata-sections -ffunction-sections -fno-exceptions";
       mkFlags = dep: langD: lib.optionals (targetPlatform != hostPlatform && dep != null && !langD) ([
-        "-O2 -idirafter ${lib.getDev dep}${dep.incdir or "/include"}"
+        "${if targetPlatform.isNone && targetPlatform.libc == "newlib-nano" then nanoFlags else commonFlags}"
```

Comment:
Not familiar with `stdenv`, but wouldn't `libc` be empty if `targetPlatform.isNone` is true? Also you can remove the string interpolation as the if/else expr is returning a string.

Reply:
`isNone` in this case refers to the os of the GNU target triplet. It applies when we are building for freestanding usage. Noted about the string interpolation, it is a leftover from experimenting.